### DISCUSSION TITLE
Accessing JSON keys

### DISCUSSION THREAD

**Discussion by GevinHasmitha** [lodash/lodash#5810] [Q&A]
I want to access and edit a nested key in the json, not the value. I tried unsetting the key value pair and setting a new pair with the key changed but this will add the new pair to the end of the json,  i want to preserver the order of the json.

Is this supported?

**Comment by juanlet** [comment]
Yes, Lodash provides methods to manipulate nested keys in an object while preserving the order. One approach is to use the _.unset() function to remove the nested key, and then use the _.set() function to set a new value with the desired key in the same position.

Here's an example:

```
const _ = require('lodash');

// Sample JSON object
const json = {
  level1: {
    level2: {
      key1: 'value1',
      key2: 'value2'
    }
  }
};

// Remove the nested key
_.unset(json, 'level1.level2.key1');

// Set a new value with the desired key at the same position
_.set(json, 'level1.level2.newKey', 'newValue');

console.log(json);
```

which will output

```
{
  level1: {
    level2: {
      key2: 'value2',
      newKey: 'newValue'
    }
  }
}
```
